### PR TITLE
Added external view key data migration for existing invoices

### PR DESCRIPTION
**Pull request by aniket-k-kaushik**
- [x] Bug fix (non-breaking change which fixes an issue)

Resolve error in staging | undefined method `external_view_key'

### CODE REVIEW

File: db/data/20220615092122_add_external_view_key_to_invoices.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+require_relative("./verifiers/invoice_default_external_view_key_verifier.rb")
+
+class AddExternalViewKeyToInvoices < ActiveRecord::Migration[7.0]
+  def up
+    Invoice.find_each do |invoice|
+      invoice.external_view_key = "#{SecureRandom.hex}"
+      invoice.save!
+    end
+    InvoiceDefaultExternalViewKeyVerifier
```
[the remark this answers is not in the document]
Comment (aniket-k-kaushik) [reply db/data/20220615092122_add_external_view_key_to_invoices.rb]: done

File: db/data/verifiers/invoice_default_external_view_key_verifier.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class InvoiceDefaultExternalViewKeyVerifier
+  def self.verify!
+    Invoice.find_each do |invoice|
+      unless invoice.external_view_key?
+        raise "Invoice external key is present for Invoice ID #{invoice.id}"
```
Comment (keshavbiswa) [inline db/data/verifiers/invoice_default_external_view_key_verifier.rb:7]: The messaging here is weird, If the invoice.external_view_key is not present, we want to raise an error with something like "external_key is not populated for Invoice with id #{invoice.id}"

File: db/data/verifiers/invoice_default_external_view_key_verifier.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+      end
+    end
+
+    puts "All invoice external key is not present"
```
Comment (keshavbiswa) [inline db/data/verifiers/invoice_default_external_view_key_verifier.rb:11]:
```suggestion
    puts "InvoiceDefaultExternalViewKey data migration verified successfully!"
```

**Review by keshavbiswa — CHANGES_REQUESTED** [review]
Requested changes with the messaging.

**Review by akhilgkrishnan — APPROVED** [review]
LGTM. Please verify manually the changes are reflected in your local or not. You can run the data migration by running ' rails data:migrate'

**Comment by aniket-k-kaushik** [comment]
> LGTM. Please verify manually the changes are reflected in your local or not. You can run the data migration by running ' rails data:migrate'

yes, it' working

**Pull request merged**